### PR TITLE
Edit target user profile and use cases

### PR DESCRIPTION
**Pull request by samuelmui8**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -257,7 +257,9 @@ _{Explain here how the data archiving feature will be implemented}_
 
 **Target user profile**:
 
-* has a need to manage a significant number of contacts
+* student foodie from NUS
```
Comment (teoks0199) [inline docs/DeveloperGuide.md:260]: Very clear user profiles 👍

**Review by teoks0199 — APPROVED** [review]

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -283,37 +285,89 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
   Use case ends.
 
 * 3a. The given index is invalid.
 
-    * 3a1. AddressBook shows an error message.
+    * 3a1. FoodNotes shows an error message.
 
       Use case resumes at step 2.
 
+**Use case: Delete a review from stall**
```
Comment (Ruizhi2001) [inline docs/DeveloperGuide.md:313]: Detailed and elaborate use cases 💯

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (jamesebond) [reply docs/DeveloperGuide.md:260]: yes lgtm

**Review by jamesebond — APPROVED** [review]

**Review by Ruizhi2001 — APPROVED** [review]
if no merge conflicts LGTM

**Pull request merged**